device-data/linking-events.md

```diff
@@ -1 +1,239 @@
+  "bgTarget": {
+    "target": 4.718135792388703,
+    "high": 8.048584587016023
+  },
+  "bolus": "hbnktcpbppt98g2a2o61qdc25k6kvo13",
+  "carbInput": 109,
+  "insulinCarbRatio": 6,
+  "insulinOnBoard": 20.032,
+  "insulinSensitivity": 0.33304487946273204,
+  "recommended": {
+    "carb": 18.25,
+    "correction": 5.25,
+    "net": 3.5
+  },
+  "units": "mmol/L",
+  "clockDriftOffset": 0,
+  "conversionOffset": 0,
+  "deviceId": "DevId0987654321",
+  "deviceTime": "2016-06-14T10:52:45",
+  "time": "2016-06-14T17:52:45.845Z",
+  "timezoneOffset": -420,
+  "uploadId": "SampleUploadId",
+  "_groupId": "f4c834c27a",
+  "guid": "b6a3e6b7-d44f-44bb-810e-189b8f2b03c2",
+  "id": "aiipljcqrquq96pugkbr2bfj4mb4r2mv",
+  "createdTime": "2016-06-14T18:03:50.664Z",
+  "_version": 0,
+  "_active": true,
+  "_schemaVersion": 1
+}]
+```
+
+### Linking events through the new platform ingestion API
+
+Unlike the legacy jellyfish ingestion service, the new platform ingestion API creates GUIDs for the `id` of each event. This means that linked events such as pairs of `wizard` and `bolus` events **cannot** be uploaded separately. Instead, only the "outer" event–that is, the event that embeds another event that it wishes to link to–is uploaded with the "inner" linked event embedded inside it. Upon ingestion of this compound event, the new platform ingestion service:
+
+- generates the GUID for the inner event and stores it on that event's `id` field
+- extracts and stores the inner event (augmented with its newly-generated `id`) as an independent event
+- updates the appropriate field on the outer event to contain just the newly-generated GUID for the inner event instead of the entire object
+
+To demonstrate with the same example data as above, the data for ingestion via the new platform ingestion API looks like:
+
+```json
+[{
+  "type": "wizard",
+  "bgInput": 32,
+  "bgTarget": {
+    "target": 85,
+    "high": 145
+  },
+  "bolus": {
+    "type": "bolus",
+    "subType": "normal",
+    "normal": 1,
+    "clockDriftOffset": 0,
+    "conversionOffset": 0,
+    "deviceId": "DevId0987654321",
+    "deviceTime": "2016-06-14T10:52:45",
+    "time": "2016-06-14T17:52:45.845Z",
+    "timezoneOffset": -420,
+    "uploadId": "SampleUploadId"
+  },
+  "carbInput": 109,
+  "insulinCarbRatio": 6,
+  "insulinOnBoard": 20.032,
+  "insulinSensitivity": 6,
+  "recommended": {
+    "carb": 18.25,
+    "correction": 5.25,
+    "net": 3.5
+  },
+  "units": "mg/dL",
+  "clockDriftOffset": 0,
+  "conversionOffset": 0,
+  "deviceId": "DevId0987654321",
+  "deviceTime": "2016-06-14T10:52:45",
+  "time": "2016-06-14T17:52:45.845Z",
+  "timezoneOffset": -420,
+  "uploadId": "SampleUploadId"
+}]
+```
+
+And the resulting data is nearly exactly the same, the only difference being that the `id`s on each object are [RFC 4122](https://tools.ietf.org/html/rfc4122 'RFC 4122: A Universally Unique IDentifier (UUID) URN Namespace') version 4 UUIDs with all `-`s stripped:
```

Comment:
Re: point 1, I'm not sure what your confusion is. The result in the database (two separate records - a `bolus` and a `wizard`) is actually the same in the jellyfish system vs. the new platform API system. The number of records ingested is different, but how the records are linked and the total number of stored records is the same... Unless I'm misunderstanding how the new platform API works?

Re: point 2, this used to bug me, but the more I started reading about UUIDs and GUIDs I realized the literature pretty much uses the terms interchangeably. If you can't beat 'em, join 'em? If you have a preference for term, I can standardize...